core/src/main/scala/anorm/Column.scala

```diff
@@ -12,16 +12,87 @@ import scala.util.{ Failure, Success => TrySuccess, Try }
 
 import resource.managed
 
-/** Column mapping */
+/**
+ * Column mapping
+ *
+ * @define mapDescription If the column is successfully parsed, then apply the given function on the result.
+ */
 @annotation.implicitNotFound(
   "No column extractor found for the type ${A}: `anorm.Column[${A}]` required; See https://github.com/playframework/anorm/blob/master/docs/manual/working/scalaGuide/main/sql/ScalaAnorm.md#column-parsers")
-trait Column[A] extends ((Any, MetaDataItem) => MayErr[SqlRequestError, A])
+trait Column[A] extends ((Any, MetaDataItem) => Either[SqlRequestError, A]) { parent =>
+
+  /**
+   * $mapDescription
+   *
+   * {{{
+   * import anorm.{ Column, SqlParser, SQL }
+   *
+   * sealed trait MyEnum
+   * case object Foo extends MyEnum
+   * case object Bar extends MyEnum
+   *
+   * val myEnumCol: Column[MyEnum] = Column.of[Int].mapResult {
+   *   case 1 => Right(Foo) // `Right` means successful
+   *   case 2 => Right(Bar)
+   *   case _ => Left(SqlMappingError("Unexpected"))
+   * }
+   *
+   * def find(id: String) =
+   *   SQL"SELECT enum_code FROM my_table WHERE id = $id".
+   *     as(SqlParser.scalar(myEnumCol).single)
+   * }}}
+   */
+  final def mapResult[B](f: A => Either[SqlRequestError, B]): Column[B] =
+    Column[B] { (v: Any, m: MetaDataItem) => parent(v, m).right.flatMap(f) }
+
+  /**
+   * $mapDescription
+   *
+   * {{{
+   * import anorm.{ Column, SqlParser, SQL }
+   *
+   * sealed trait MyEnum
+   * case object Foo extends MyEnum
+   * case object Bar extends MyEnum
+   *
+   * val myEnumCol: Column[MyEnum] = Column.of[Int].map {
+   *   case 1 => Right(Foo) // `Right` means successful
+   *   case 2 => Right(Bar)
+   * }
+   *
+   * def find(id: String) =
+   *   SQL"SELECT enum_code FROM my_table WHERE id = $id".
+   *     as(SqlParser.scalar(myEnumCol).single)
+   * }}}
+   */
+  def map[B](f: A => B): Column[B] = mapResult[B] { a =>
+    try {
+      Right(f(a))
+    } catch {
+      case cause: Throwable => Left(SqlRequestError(cause))
```

Comment:
All of these catch blocks should only catch `NonFatal(cause)`. If it's a fatal error, I would expect the error to bubble up. I believe this is what most Scala users expect since this is what `Try` does.